pkg/logging/controller/logging/clusterlogging.go

```diff
@@ -0,0 +1,92 @@
+package logging
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	loggingconfig "github.com/rancher/rancher/pkg/logging/config"
+	"github.com/rancher/rancher/pkg/logging/generator"
+	"github.com/rancher/rancher/pkg/logging/utils"
+)
+
+type ClusterLoggingSyncer struct {
+	management           config.ManagementContext
+	clusterLoggingClient v3.ClusterLoggingInterface
+	projectLoggingClient v3.ProjectLoggingInterface
+	K8sClient            kubernetes.Interface
+}
+
+func RegisterClusterLogging(cluster *config.ClusterContext) {
+	clusterloggingClient := cluster.Management.Management.ClusterLoggings("")
+	syncer := &ClusterLoggingSyncer{
+		management:           *cluster.Management,
+		K8sClient:            cluster.K8sClient,
+		clusterLoggingClient: clusterloggingClient,
+		projectLoggingClient: cluster.Management.Management.ProjectLoggings(""),
+	}
+
+	clusterloggingClient.AddClusterScopedHandler("cluster-logging-controller", cluster.ClusterName, syncer.Sync)
+}
+
+func (c *ClusterLoggingSyncer) Sync(key string, obj *v3.ClusterLogging) error {
+	if utils.GetClusterTarget(obj.Spec) == "embedded" {
+		err := utils.CreateEmbeddedTarget(c.K8sClient, loggingconfig.LoggingNamespace)
+		if err != nil {
+			logrus.Errorf("create cluster logging embedded target error %v", err)
+			return err
+		}
+	} else {
+		utils.RemoveEmbeddedTarget(c.K8sClient, loggingconfig.LoggingNamespace)
```

Comment:
RemoveEmbeddedTarget should return an error, don't just log it internally. Then in the sync we can decide whether to ignore it, or simply log it.